pkg/k8s/utils.go

```diff
@@ -0,0 +1,57 @@
+package k8s
+
+import (
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+// isClusterEnvironment checks if the program is running in a Kubernetes cluster.
+func isClusterEnvironment() bool {
+	return fileExists(tokenPath) && fileExists(certPath)
+}
+
+func fileExists(path string) bool {
+	_, err := os.Stat(path)
+	return err == nil
+}
+
+// getClusterConfig returns the appropriate Kubernetes cluster configuration.
+func getClusterConfig() (*rest.Config, error) {
+	if isClusterEnvironment() {
+		return rest.InClusterConfig()
+	}
+
+	// build the configuration from the kubeconfig file
+	kubeconfig := filepath.Join(os.Getenv("HOME"), ".kube", "config")
+	return clientcmd.BuildConfigFromFlags("", kubeconfig)
+}
+
+// precompile the regular expression to avoid recompiling it on every function call
+var invalidCharsRegexp = regexp.MustCompile(`[^a-z0-9-]+`)
+
+// SanitizeName ensures compliance with Kubernetes DNS-1123 subdomain names. It:
```

Comment:
is an empty string a valid name? if not we might want to panic on `len(sanitized) == 0`

Reply:
Nice catch. will add it. However, not sure if we should panic. we might want to add a random string instead of an empty one. Not really a big fan of panic for this simple thing.
@jrmanes @smuu what do you think?

Reply:
we could also return a string like this: `empty-name`

Reply:
well in theory yes, we might. In practice, the output of this function goes to another func that generates a name with a random suffix. If we call that one, then we need to return an error here.

Reply:
yes that makes sense